### PR TITLE
Inline GeometryInfo<dim>::line_to_cell_vertices.

### PR DESCRIPTION
**Pull request by drwells**
This provides about a 5% speed boost in `TransfiniteInterpolationManifold` since the compiler can rewrite the loop over the line vertices.

### CODE REVIEW

File: include/deal.II/base/geometry_info.h [hunk 1]
```diff
@@ -2804,6 +2805,71 @@ GeometryInfo<3>::is_inside_unit_cell (const Point<3> &p,
...
+
+
+template <>
+inline
+unsigned int
+GeometryInfo<4>::line_to_cell_vertices (const unsigned int,
+                                        const unsigned int)
+{
+  Assert(false, ExcNotImplemented());
+  // TODO use invalid_unsigned_int
```
Comment (tjhei) [inline include/deal.II/base/geometry_info.h]: why did you change it to -1 and leave a todo?

File: include/deal.II/base/geometry_info.h [hunk 2]
```diff
@@ -1729,6 +1729,7 @@ struct GeometryInfo
    * face_to_cell_vertices() function.
    */
   static
+  inline
```
Comment (kronbichler) [inline include/deal.II/base/geometry_info.h]: Why did you need to put the `inline` statement at the place of the declaration? Our convention in deal.II is to not put the `inline` here (except for `DEAL_II_ALWAYS_INLINE`). Actually, for templated classes/functions it is not even needed at the place of the definition - even though we still use that convention.

File: include/deal.II/base/geometry_info.h (hunk 1 above)
Comment (drwells) [inline include/deal.II/base/geometry_info.h]: At some point I couldn't get things to compile with `invalid_unsigned_int`; I forgot to go back and fix this. I'll do it in a moment.

File: include/deal.II/base/geometry_info.h (hunk 2 above)
Comment (drwells) [inline include/deal.II/base/geometry_info.h]: I completely missed that we used that style. I'll fix it.

**Comment by drwells** [comment]
@kronbichler thanks :)
